rdr_service/genomic/genomic_job_components.py

```diff
@@ -1036,58 +1049,71 @@ def create_new_genomic_participants(self, from_date):
         :return: result
         """
         samples = self._get_new_biobank_samples(from_date)
+        samples_meta = self.GenomicSampleMeta(*samples)
+
         if len(samples) > 0:
-            return self.process_samples_into_manifest(samples, cohort=self.COHORT_3_ID)
+            return self.process_samples_into_manifest(samples_meta, cohort=self.COHORT_3_ID)
 
         else:
             logging.info(f'New Participant Workflow: No new samples to process.')
             return GenomicSubProcessResult.NO_FILES
 
     def create_c2_genomic_participants(self, from_date, local=False):
         """
-        This method determines which samples to enter into the genomic system
-        from Cohort 2.
+        Creates Cohort 2 Participants in the genomic system using reconsent.
         Validation is handled in the query that retrieves the newly consented
-        participants' samples to process.
-        :param: from_date : the date from which to lookup new biobank_ids
+        participants. Only valid participants are currently sent.
+        Refactored to first pull valid participants, then pull their samples,
+        applying the new business logic of prioritizing
+        collection date & blood over saliva.
+
+        :param: from_date : the date from which to lookup new participants
         :return: result
         """
-        samples = self._get_new_c2_consent_samples(from_date)
 
-        if len(samples) > 0:
-            return self.process_samples_into_manifest(samples, cohort=self.COHORT_2_ID, local=local)
+        participants = self._get_new_c2_participants(from_date)
+        participant_matrix = self.GenomicSampleMeta(*participants)
+
+        if len(participants) > 0:
+            for i, _bid in enumerate(participant_matrix.bids):
+                sample_data = self._get_usable_sample_from_participant(pid=participant_matrix.pids[i],
+                                                                       bid=_bid)
+
+                # update the sample id and collected site
+                if sample_data != (None, None):
```

Comment:
Would this match an empty return or should there be another None?

Reply:
Good catch, the biobank order id was a late addition. I may change it to the following since all values are required:
`if None not in sample_data:`